passes/sat/qbfsat.cc

```diff
@@ -187,113 +199,146 @@ void write_solution(RTLIL::Module *module, const QbfSolutionType &sol, const std
 	if (!fout)
 		log_cmd_error("could not open solution file for writing.\n");
 
-	dict<std::string, std::string> hole_loc_to_name = get_hole_loc_name_map(module, sol);
-	for(auto &x : sol.hole_to_value)
-		fout << hole_loc_to_name[x.first] << "=" << x.second << std::endl;
+	//There is a question here: How exactly shall we identify holes?
+	//There are at least two reasonable options:
+	//1. By the source location of the $anyconst cells
+	//2. By the name(s) of the wire(s) connected to each SigBit of the $anyconst cell->getPort(ID::Y) SigSpec.
+	//
+	//Option 1 has the benefit of being very precise.  There is very limited potential for confusion, as long
+	//as the source attribute has been set.  However, if the source attribute is not set, this won't work.
+	//More importantly, we want to have the ability to port hole assignments to other modules with compatible
+	//hole names and widths.  Obviously in those cases source locations of the $anyconst cells will not match.
+	//
+	//Option 2 has the benefits previously described, but wire names can be changed automatically by 
+	//optimization or techmapping passes, especially when (ex/im)porting from BLIF for optimization with ABC.
```

Comment:
Would the `hdlname` attribute help here?

Reply:
I'm not sure.

Maybe it could be used in place of source locations?  But source locations are still quite a good way to track an object; I'm not sure matching by `hdlname` gives any advantage.

I don't think it can replace regular names, with which we already first try to match: The point of name matching is to allow the use of the same hole assignments found by a solver to specialize a different module with compatible holes.  I understand the `hdlname` attribute is rightfully "sticky" and e.g. does not update when wires are renamed (right? please correct me if wrong), so matching with it might impede the kind of flexibility that name matching is trying to allow.  And if it doesn't impede flexibility, then it's not clear to me how it differs from regular name matching.

Reply:
I can't think of any specific benefit it brings.  I'll keep it in mind and who knows, maybe a benefit will become clear at some point.  But for now at least, I think I'll leave it as it is.